### ISSUE TITLE
Maybe require #[repr(C)]?

### ISSUE THREAD

**Issue by bjorn3** [CorentinDeblock/wrld#1]
```
struct Test {
    position: [f32; 2],
    color: [f32; 4]
}
```
doesn't have a fixed layout. For example rustc may put color before position or add padding. In fact it will in all likelyhood if you use `-Zrandomize-layout`.

**Comment by CorentinDeblock** [comment]
You are correct, i will put patch asap. Probably in one or two week max.

But be aware that wrld::Desc macro generate only the VertexBufferLayout of the corresponding layout structure. It's up to you to format data with bytemuck and put the #[repr(C)] macro on your structure. Still an issue anyways.

Also you can have a look at BufferData derive macro which generate a structure with bytemuck and repr(C) macro with all the corresponding conversion to one and another.

**Comment by bjorn3** [comment]
Makes sense. I missed the BufferData derive macro.

**Comment by CorentinDeblock** [comment]
It's ok, he was released with 0.2.0 which was just released today.

**Comment by CorentinDeblock** [comment]
This issue has been fix.

Wrld::desc now need #[repr(C)] or #[repr(transparent)] in 0.2.2, i will close this issue.
